libvncserver/main.c

```diff
@@ -560,10 +560,14 @@ clientInput(void *data)
 	tv.tv_sec = 60; /* 1 minute */
 	tv.tv_usec = 0;
 
+retry:
 	n = select(nfds + 1, &rfds, &wfds, &efds, &tv);
 
 	if (n < 0) {
 	    rfbLogPerror("ReadExact: select");
+        if (errno == EINTR) {
+            goto retry;
```

Comment:
why wouldn't a simple `continue` work as in the other cases?

Reply:
@piec the goto itself is not an issue, the problem is if the client disconnect.
In that case the FD will become invalid, but your retry will cut the check done some line before, and it will never finish till it'll probably crash.  
So please try with the `continue`, I cannot reproduce the problem to test.